### ISSUE TITLE
Random string generation broken in last version

### ISSUE THREAD

**Issue by SvanT** [microsoft/pyright#3779]
This is valid python code that worked in version 1.1.264 but not in 1.1.265

```
import random
import string

"".join([random.choice(string.ascii_letters) for _ in range(8)])
```

The error is:
```
error: Argument of type "LiteralString" cannot be assigned to parameter "seq" of type "SupportsLenAndGetItem[_T@choice]" in function "choice"
    "str" is incompatible with protocol "SupportsLenAndGetItem[_T@choice]"
      TypeVar "_T_co@SupportsLenAndGetItem" is covariant
        Type "str" cannot be assigned to type "LiteralString"
          "str" is incompatible with "LiteralString" (reportGeneralTypeIssues)
```

**Comment by erictraut** [comment]
Thanks for the bug report. This was a long-standing bug that was exposed by two different (seemingly unrelated) pair of changes. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.266, which I just published. It will also be included in a future release of pylance.
